### ISSUE TITLE
[JENKINS-30901] Use the name set by the 'Build Name Setter' plugin no longer working in JIRA updates

### ISSUE THREAD

**Issue by jenkins-infra-bot** [jenkinsci/jira-plugin#941]
<p>We recently upgraded to Jenkins version. 1.627 and JIRA plugin version 2.0 and noticed a change in behavior I cannot find in any changelogs or bugs so far.</p>

<p>We use the Build Name Setter plugin to set the name of a build to the tag of the code being build.<br/>
For example build #245 would be named "Product_Number_Release".<br/>
Before the update, the JIRA plugin would create comments like this:</p>
<blockquote><p>SUCCESS: Integrated in  JOBNAME #Tagname</p></blockquote>
<p>After the update this changed to:</p>
<blockquote><p>SUCCESS: Integrated in  JOBNAME #Buildnumber</p></blockquote>

<p>Example:<br/>
Before: "SUCCESS: Integrated in  Release_Main #Main_255_Release"<br/>
After: "SUCCESS: Integrated in  Release_Main #345"</p>

<p>It looks like a very minor issue and I first suspected that I simply would have to set a new parameter somewhere to restore the old behavior but wasn't able to find it. </p>

<p>As we are using a dedicated publishing job from different build jobs, the build number of the publishing job is of no use to us. <br/>
Please advise how to get back to build names.</p>

<p>Thanks<br/>
Paul</p>

---
<details><summary><i>Originally reported by <img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/paul_muller.jpg" title="paul_muller's avatar" /> <a href="https://issues.jenkins.io/secure/ViewProfile.jspa?name=paul_muller">paul_muller</a>, imported from: <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-30901" target="_blank">Use the name set by the 'Build Name Setter' plugin no longer working in JIRA updates</a></i></summary>
<i><ul>
<li><b>status</b>: Closed
<li><b>priority</b>: Minor
<li><b>component(s)</b>: jira-plugin
<li><b>label(s)</b>: confirmed
<li><b>resolution</b>: Cannot Reproduce
<li><b>resolved</b>: 2025-05-27T09:35:07+00:00
<li><b>votes</b>: 1
<li><b>watchers</b>: 1
<li><b>imported</b>: 2025-12-16
</ul></i>
<details><summary>Raw content of original issue</summary>

<pre>
<p>We recently upgraded to Jenkins version. 1.627 and JIRA plugin version 2.0 and noticed a change in behavior I cannot find in any changelogs or bugs so far.</p>

<p>We use the Build Name Setter plugin to set the name of a build to the tag of the code being build.
For example build #245 would be named "Product_Number_Release".
Before the update, the JIRA plugin would create comments like this:</p>
<blockquote><p>SUCCESS: Integrated in  JOBNAME #Tagname</p></blockquote>
<p>After the update this changed to:</p>
<blockquote><p>SUCCESS: Integrated in  JOBNAME #Buildnumber</p></blockquote>

<p>Example:
Before: "SUCCESS: Integrated in  Release_Main #Main_255_Release"
After: "SUCCESS: Integrated in  Release_Main #345"</p>

<p>It looks like a very minor issue and I first suspected that I simply would have to set a new parameter somewhere to restore the old behavior but wasn't able to find it. </p>

<p>As we are using a dedicated publishing job from different build jobs, the build number of the publishing job is of no use to us. 
Please advise how to get back to build names.</p>

<p>Thanks
Paul</p></pre>
</details>
</details>
<details><summary><i>environment</i></summary>

```
Jenkins version . 1.627<br/>
JIRA plugin version 2.0<br/>
```
</details>

**Comment by jenkins-infra-bot** [comment]
<details><summary><i><img align="left" width="20" src="https://raw.githubusercontent.com/jenkinsci/artifacts-from-jira-issues/refs/heads/main/avatars/warden.png" title="warden's avatar" /> warden:</i></summary>

<ul>
<li><i>Original <a class="original-jira-link" href="https://issues.jenkins.io/browse/JENKINS-30901?focusedId=454634&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-454634">comment link</a></i>
<li><details><summary><i>Raw content of original comment:</i></summary>

<pre>
<p>This should have been fixed in the past - if it's still happening, please create a new issue at <a href="https://github.com/jenkinsci/jira-plugin/issues" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jira-plugin/issues</a> with current configuration details.</p>
</pre>
</details>

</ul>
</details>

<p>This should have been fixed in the past - if it's still happening, please create a new issue at <a href="https://github.com/jenkinsci/jira-plugin/issues" class="external-link" target="_blank" rel="nofollow noopener">https://github.com/jenkinsci/jira-plugin/issues</a> with current configuration details.</p>
